src/build_helper/src/fs/mod.rs

```diff
@@ -31,25 +31,47 @@ where
 ///
 /// FIXME: this implementation is insufficiently robust to replace bootstrap's clean `rm_rf`
 /// implementation:
-///
-/// - This implementation currently does not perform retries.
 #[track_caller]
 pub fn recursive_remove<P: AsRef<Path>>(path: P) -> io::Result<()> {
     let path = path.as_ref();
-    let metadata = fs::symlink_metadata(path)?;
-    #[cfg(windows)]
-    let is_dir_like = |meta: &fs::Metadata| {
-        use std::os::windows::fs::FileTypeExt;
-        meta.is_dir() || meta.file_type().is_symlink_dir()
-    };
-    #[cfg(not(windows))]
-    let is_dir_like = fs::Metadata::is_dir;
 
-    if is_dir_like(&metadata) {
-        fs::remove_dir_all(path)
-    } else {
-        try_remove_op_set_perms(fs::remove_file, path, metadata)
+    for retry in 0..5 {
+        match fs::symlink_metadata(path) {
```

Comment:
Why would this need to be retried? AFAIK the reason for the retries is that antivirus sometimes prevents deletion of files due to keeping them open while scanning. Reading a symlink doesn't suffer from that AFAIK.

Reply:
Indeed, https://github.com/rust-lang/rust/pull/139870#discussion_r2044714032 above :D